tools/benchmark/src/Reporter.ts

```diff
@@ -39,25 +39,14 @@ import * as path from "node:path";
 import chalk from "chalk";
 import Table from "easy-table";
 
-import {
-	isResultError,
-	type BenchmarkData,
-	type BenchmarkResult,
-	type CustomData,
-} from "./ResultTypes";
+import { isResultError, type BenchmarkData, type BenchmarkResult } from "./ResultTypes";
 import { pad, prettyNumber } from "./RunnerUtilities";
-import { ExpectedCell, addCells, numberCell, stringCell } from "./resultFormatting";
 
 interface BenchmarkResults {
 	table: Table;
 	benchmarksMap: Map<string, Readonly<BenchmarkResult>>;
 }
 
-const expectedKeys: ExpectedCell[] = [
-	stringCell("error", "error", (message) => chalk.red(message || "Error")),
```

Comment:
Sorry @jikim-msft , I realized one more thing 😅 . In the current code, if a test fails, the fact that we have "error" in the expected keys causes the error to be printed to the console too as a new column (that only the failed tests have a value for).

We should keep that too, otherwise we won't see what the actual errors were for the failing tests. For that we can make a few small changes in `recordTestResult`. When we check if `isResultError(result)`, when it is, we should create table cells for status, name, elapsed seconds, and error (`result.error` will be available in that case), and nothing more. And if `result` is not an error, then do what we do today (status, name, elapsed seconds, then all the custom data).

Other than this, I think the rest of the PR looks good.

Reply:
Is there a recommended ordering for logging outputs? 

I think the ideal ordering of the console output should be something like (at least the `name` should come before the `error`):
`status`, `name`, `total time (s)`, `error`

```ts
if (isResultError(result)) {
	table.cell("status", `${pad(4)}${chalk.red("×")}`);
	table.cell("error", result.error);
} else {
	table.cell("status", `${pad(4)}${chalk.green("✔")}`);
}
table.cell("name", chalk.italic(testName));
table.cell("total time (s)", prettyNumber((result as BenchmarkData).elapsedSeconds, 2));
```

With the current code, the `error` will appear right after the `status`, which I believe is unintuitive. I could instead remove the `error` in the first if-statement and create another `if (isResultError(result))` after creating the `name` and `total time (s)` cell, but this also feels repetitive to me. 

Curious to know how you think about the ordering of `status`, `error`, `name`, `total time (s)`.

Reply:
I like what you suggested, status, name, total time, error. But I'd probably go with a single if, and create all the required cells on each branch. Slightly repetitive, but I think cleaner than and if at the beginning for the status, then common code, then another if.